raiden/_monkey.py

```diff
@@ -0,0 +1,33 @@
+import asyncio.selector_events
+
+import aioice.ice
+
+
+def _patch_aioice() -> None:
+    # local_candidate could be None so make sure we check for it before
+    # invoking data_received
+    def StunProtocol__connection_lost(self, exc: Exception) -> None:  # type: ignore
+        self._StunProtocol__log_debug("connection_lost(%s)", exc)
+        if self.local_candidate is not None:
+            self.receiver.data_received(None, self.local_candidate.component)
+        self._StunProtocol__closed.set_result(True)
```

Comment:
shouldnt this be `self.__closed.set_result(True)`  or is this some python native thing?

Reply:
and what is the reasoning behing setting the result anyway?

Reply:
> shouldnt this be `self.__closed.set_result(True)` or is this some python native thing?

Not really, due to how Python's name mangling works. See e.g. https://docs.python.org/3/tutorial/classes.html#private-variables.

Setting the result is what the the original method in `aioice` does and it is very much needed.
The only functional change here is actually the addition of the `if` line.